src/test_driver/linux-cirque/helper/CHIPTestBase.py

```diff
@@ -0,0 +1,287 @@
+import cirque.proto.service_pb2_grpc as service_pb2_grpc
+
+from cirque.proto.device_pb2 import DeviceSpecification
+from cirque.common.cirquelog import CirqueLog
+from cirque.common.utils import sleep_time
+
+from cirque.proto.capability_pb2 import (
+    WeaveCapability, ThreadCapability, WiFiCapability, XvncCapability,
+    InteractiveCapability, LanAccessCapability
+)
+
+class CHIPVirtualHome:
+    def __init__(self, device_config):
+        self.home_id = None
+        self.device_config = device_config
+        self.device_ids = set()
+        self.devices = []
+        self.non_ap_devices = []
+        self.ap_devices = []
+    
+    def sequenceMatch(self, string, patterns):
+        last_find = 0
+        for s in patterns:
+            self.logger.info('Finding {}'.format(s))
+            this_find = string.find(s, last_find)
+            if this_find < 0:
+                self.logger.info('Cannot find!')
+                return False
+            self.logger.info("Found at index={}".format(this_find))
+            last_find = this_find + len(s)
+        return True
+
+    def assertTrue(self, exp, note = None):
+        if not exp:
+            if note:
+                self.logger.error(note)
+            raise RuntimeError
+    
+    def assertEqual(self, val1, val2, note = None):
+        if not val1 == val2:
+            if note:
+                self.logger.error(note)
+            raise RuntimeError
+
+    def assertNotEqual(self, val1, val2, note = None):
+        if val1 == val2:
+            if note:
+                self.logger.error(note) 
+            raise RuntimeError
+
+    def tearDownClass(self):
+
+        self.logger.info("tearing down test class")
+        self.logger.info("stopping home: {}".format(self.home_id))
+        self.stub.StopCirqueHome(service_pb2.StopCirqueHomeRequest(
+                home_id=self.home_id))
+
+        self.channel.close()
+
+    def setUpClass(self):
+        self.channel = grpc.insecure_channel('localhost:50051')
+        self.stub = service_pb2_grpc.CirqueServiceStub(self.channel)
+        
+        if hasattr(self, 'logger') and self.logger:
+            return
+        self.logger = CirqueLog.get_cirque_logger('CHIPVirtualHome')
+        sh = logging.StreamHandler()
+        sh.setFormatter(
+            logging.Formatter(
+                '%(asctime)s [%(name)s] %(levelname)s %(message)s'))
+        self.logger.addHandler(sh)
+    
+    def create_home(self):
+        home_id = self.stub.CreateCirqueHome(
+            service_pb2.CreateCirqueHomeRequest()).home_id
+
+        self.logger.info("home id: {} created!".format(home_id))
+
+        self.assertTrue(
+            home_id in self.stub.ListCirqueHomes(
+                service_pb2.ListCirqueHomesRequest()).home_id,
+            "created home_id could not find in the cirque service!!")
+        
+        self.home_id = home_id
+
+        device_ids = set()
+
+        for device in self.device_config:
```

Comment:
could use `for device in self.device_config.values()` and use `**device` instead of `**self.device_config[device]`

Reply:
No longer applied since we switched to flask api